fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/RewriteMultiDistinctByCTERule.java

```diff
@@ -137,14 +137,20 @@ public boolean check(OptExpression input, OptimizerContext context) {
             return false;
         }
 
-        if (agg.getLimit() >= 0) {
-            return false;
-        }
 
         List<CallOperator> distinctAggOperatorList = agg.getAggregations().values().stream()
                 .filter(CallOperator::isDistinct).collect(Collectors.toList());
         boolean hasMultiColumns = distinctAggOperatorList.stream().anyMatch(f -> f.getChildren().size() > 1);
 
+        if (agg.hasSkew() && distinctAggOperatorList.size() > 1 && !hasMultiColumns &&
```

Comment:
In 2 new added UT, the condition [distinctAggOperatorList.size() > 1] is always false. Would you mind to add more UT for this?

Reply:
distinctAggOperatorList.size() > 1 is not always false,  Rule is applied to Logical Plan when the Rule match plan and Rule.check return true.